`configure_mapper` function missing return type annotation

### Ensure stubs packages are not installed

- [X] No sqlalchemy stub packages is installed (both `sqlalchemy-stubs` and `sqlalchemy2-stubs` are not compatible with v2)

### Verify if the api is typed

- [X] The api is not in a module listed in [#6810](https://github.com/sqlalchemy/sqlalchemy/issues/6810) so it should pass type checking

### Describe the typing issue

When using Mypy as a typechecker, calls to `configure_mapper` result in a type error

### To Reproduce

```python
configure_mapper()
```


### Error

```
src/db_main.py: note: In function "set_up_pg_db":
src/db_main.py:51: error: Call to untyped function "configure_mappers" in typed context  [no-untyped-call]
        configure_mappers()
        ^~~~~~~~~~~~~~~~~~~
Found 1 error in 1 file (checked 166 source files)
``` 

### Versions

- OS: MacOS Ventura (13.5)
- Python: 3.8.5
- SQLAlchemy: 2.0.18
- Type checker (eg: mypy 0.991, pyright 1.1.290, etc): mypy 1.4.1


### Additional context

_No response_

--- a/lib/sqlalchemy/orm/mapper.py
+++ b/lib/sqlalchemy/orm/mapper.py
@@ -4097,7 +4097,7 @@ class _OptGetColumnsNotAvailable(Exception):
     pass
 
 
-def configure_mappers():
+def configure_mappers() -> None:
     """Initialize the inter-mapper relationships of all mappers that
     have been constructed thus far across all :class:`_orm.registry`
     collections.

